share/shwap/row_test.go

```diff
@@ -33,6 +34,15 @@ func TestRowFromShares(t *testing.T) {
 			}
 			require.Equal(t, half, row.halfShares)
 			require.Equal(t, side, row.side)
+
+			data, err := json.Marshal(row)
+			require.NoError(t, err)
+
+			decodedRow := &Row{}
+			err = json.Unmarshal(data, decodedRow)
+			require.NoError(t, err)
+			require.Equal(t, row.halfShares, decodedRow.halfShares)
+			require.Equal(t, row.side, decodedRow.side)
```

Comment:
It should be its own test without calling `Shares` method as its gonna cache in #4005